### DISCUSSION TITLE
next dev command begins to stall after being complied.

### DISCUSSION THREAD

**Discussion by ree-see** [vercel/next.js#14689] [Help]
# Bug report
Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
## Describe the bug
When running the `next dev` the server starts up then complies but once done the browser isn't able to reload and then stalls for about a minute and then gives the error.

`
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x100bc9ae7 node::Abort() (.cold.1) [/usr/local/Cellar/node/14.4.0/bin/node]
 2: 0x10008186d node::FatalError(char const*, char const*) [/usr/local/Cellar/node/14.4.0/bin/node]
 3: 0x1000819d6 node::OnFatalError(char const*, char const*) [/usr/local/Cellar/node/14.4.0/bin/node]
`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to 'https://github.com/jrisse29041408/daleague-web'
2. Clone the repo
3. cd into the repo 
4. run `next dev`

## Expected behavior

It should work completely fine.
## System information

- OS: macOS Catalina
- Browser Chrome
- Version of Next.js: latest
- Version of Node.js: v14.4.0

**Comment by jamesmosier** [comment]
The issue is a circular reference. `<HomepageBody />` is imported in `IsntAuthLayout` and then `HomePageBody` uses `ResponsiveContainer` and that renders `DesktopComponent` and then that imports `HomepageBody` so then the whole cycle starts over again.

Removing `<HomepageBody/>` from `components/layout/DesktopContainer.jsx` and `MobileContainer.jsx` resolves the issue.
